common/functions.sh

```diff
@@ -18,19 +18,27 @@ function ensure_root() {
   local me=$(whoami)
   if [ "$me" != 'root' ] ; then
     echo "ERROR: this script requires root:"
-    echo "       sudo -E $0"
+    echo "       mysudo -E $0"
     exit 1;
   fi
 }
 
 function ensure_port_free() {
   local port=$1
-  if sudo lsof -Pn -sTCP:LISTEN -i :$port ; then
+  if mysudo lsof -Pn -sTCP:LISTEN -i :$port ; then
     echo "ERROR: Port $port is already opened by another process"
     exit 1
   fi
 }
 
+function mysudo() {
+    if [[ $DISTRO == "macosx" ]]; then
+	"$@"
+    else
+	sudo "$@"
```

Comment:
could you check that quotes around $@ dont break calls because of sensitivity to quotes,
e.g. of the call where could be problems in linux:
mysudo docker inspect --format '{{ .NetworkSettings.Gateway }}' "$RPM_CONTAINER_NAME"

Reply:
```
ubuntu@autonomous-horntail:~$ cat /etc/debian_version 
buster/sid
ubuntu@autonomous-horntail:~$ cat test.sh
#!/bin/bash

RPM_CONTAINER_NAME=agitated_bell

function mysudo() {
    sudo "$@"
}

mysudo docker inspect --format '{{ .NetworkSettings.SandboxID }}' "$RPM_CONTAINER_NAME"
ubuntu@autonomous-horntail:~$ ./test.sh 
faf741c571f25af144e08600638d48c3c534b746e3f331e989bd6d2178209539
```